provider.go

```diff
@@ -0,0 +1,177 @@
+// Package namecheap implements a DNS record management client compatible
+// with the libdns interfaces for namecheap.
+package namecheap
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/libdns/libdns"
+
+	"github.com/libdns/namecheap/internal/namecheap"
+)
+
+func parseIntoHostRecord(record libdns.Record) namecheap.HostRecord {
+	return namecheap.HostRecord{
+		HostID:     record.ID,
+		RecordType: namecheap.RecordType(record.Type),
+		Name:       record.Name,
+		TTL:        uint16(record.TTL.Seconds()),
+		Address:    record.Value,
+	}
+}
+
+func parseFromHostRecord(hostRecord namecheap.HostRecord) libdns.Record {
+	return libdns.Record{
+		ID:    hostRecord.HostID,
+		Type:  string(hostRecord.RecordType),
+		Name:  hostRecord.Name,
+		TTL:   time.Duration(hostRecord.TTL) * time.Second,
+		Value: hostRecord.Address,
+	}
+}
+
+// Provider facilitates DNS record manipulation with namecheap.
+// The libdns methods that return updated structs do not have
+// their ID fields set since this information is not returned
+// by the namecheap API.
+type Provider struct {
+	// APIKey is your namecheap API key.
+	// See: https://www.namecheap.com/support/api/intro/
+	// for more details.
+	APIKey string `json:"api_key,omitempty"`
+
+	// User is your namecheap API user. This can be the same as your username.
+	User string `json:"user,omitempty"`
+
+	// APIEndpoint to use. If testing, you can use the "sandbox" endpoint
+	// instead of the production one.
+	APIEndpoint string `json:"api_endpoint,omitempty"`
+
+	// ClientIP is the IP address of the requesting client.
+	// If this is not set, a discovery service will be
+	// used to determine the public ip of the machine.
+	// You must first whitelist your IP in the namecheap console
+	// before using the API.
+	ClientIP string `json:"client_IP,omitempty`
```

Comment:
Usualy convention like this would dictate `client_ip` (lowercase).

Reply:
Whoops that was a mistake